### PR TITLE
buildSds() creates GoogleGrpc TargetSpecifier if TargetUri is not empty

### PR DESCRIPTION
**Pull request by bewebi**
# Description

- When generating SDS config, if `TargetUri` is specified and `ClusterName` is not specified, create a `GrpcService` with a `GoogleGrpc` `TargetSpecifier`
  - Use the `CertificateSecretName` or `ValidationContextName`, depending on which we are generating config for, as the `StatPrefix` 

# Context

This has always been expected behavior, and the fact that we do not create the expected `GoogleGrpc` `TargetSpecifier` until now is considered a bug

In theory `TargetUri` and `ClusterName` are mutually exclusive and could/should be part of the `SdsBuilder` `oneof` in the `SDSConfig` proto
In order to prevent a breaking change we opt not to modify the proto and prioritize `ClusterName` if bother are specified in order to maintain previous behavior of ignoring `TargetUri` in that case

I've manually verified that the expected config is generated given the VS from the original [issue](https://github.com/solo-io/gloo/issues/6799).

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/pkg/utils/ssl_test.go [hunk 1]
```diff
@@ -400,6 +398,32 @@ var _ = Describe("Ssl", func() {
 
 		})
 
+		When("TargetUri is specified", func() {
+			BeforeEach(func() {
+				sdsConfig.TargetUri = "targetUri"
+			})
+
+			It("should have a sds setup with a default cluster name", func() {
```
Comment (kdorosh) [inline projects/gloo/pkg/utils/ssl_test.go:406]: nit, name needs to be updated

**Review by kdorosh — COMMENTED** [review]
looks reasonable to me!

File: projects/gloo/pkg/utils/ssl_test.go [hunk 2]
```diff
@@ -400,6 +398,69 @@ var _ = Describe("Ssl", func() {
...
 		})
 
+		Context("TargetUri is specified", func() {
+			BeforeEach(func() {
+				sdsConfig.TargetUri = "targetUri"
+			})
+
+			When("only TargetUri is specified", func() {
+
+				FIt("should have a sds setup with a GoogleGrpc TargetSpecifier with the expected TargetUri", func() {
```
Comment (kdorosh) [inline projects/gloo/pkg/utils/ssl_test.go:408]: unfocus

Comment (bewebi) [reply projects/gloo/pkg/utils/ssl_test.go]: 🤦‍♂️

**Review by nfuden — APPROVED** [review]

**Review by nfuden — APPROVED** [review]

**Pull request merged**